tests/e2e/util.go

```diff
@@ -6823,3 +6827,92 @@ func getVmdkPathFromVolumeHandle(sshClientConfig *ssh.ClientConfig, masterIp str
 	vmdkPath := result.Stdout
 	return vmdkPath
 }
+
+// checkVcServicesHealthPostReboot returns waitErr, if VC services are not running in given timeout
+func checkVcServicesHealthPostReboot(ctx context.Context, host string, timeout ...time.Duration) error {
+	var pollTime time.Duration
+	// if timeout is not passed then default pollTime to be set to 30 mins
+	if len(timeout) == 0 {
+		pollTime = pollTimeout * 6
+	} else {
+		pollTime = timeout[0]
+	}
+	//list of default stopped services in VC
+	var defaultStoppedServicesList = []string{"vmcam", "vmonapi", "vmware-imagebuilder", "vmware-netdumper",
+		"vmware-rbd-watchdog", "vmware-vcha"}
+	waitErr := wait.PollUntilContextTimeout(ctx, poll, pollTime, true,
+		func(ctx context.Context) (bool, error) {
+			var pendingServiceslist []string
+			var noAdditionalServiceStopped = false
+			sshCmd := fmt.Sprintf("service-control --%s", statusOperation)
+			framework.Logf("Invoking command %v on vCenter host %v", sshCmd, host)
+			result, err := fssh.SSH(ctx, sshCmd, host, framework.TestContext.Provider)
+			if err != nil || result.Code != 0 {
+				fssh.LogResult(result)
+				return false, fmt.Errorf("couldn't execute command: %s on vCenter host: %v", sshCmd, err)
+			}
+			framework.Logf("Command %v output is %v", sshCmd, result.Stdout)
+			// getting list of services which are stopped currently
+			services := strings.SplitAfter(result.Stdout, svcRunningMessage+":")
+			servicesStoppedInVc := strings.Split(services[1], svcStoppedMessage+":")
+			// getting list if pending services if any
+			if strings.Contains(servicesStoppedInVc[0], "StartPending:") {
+				pendingServiceslist = strings.Split(servicesStoppedInVc[0], "StartPending:")
+				pendingServiceslist = strings.Split(strings.Trim(pendingServiceslist[1], "\n "), " ")
+				framework.Logf("Additional service %s in StartPending state", pendingServiceslist)
+			}
+			servicesStoppedInVc = strings.Split(strings.Trim(servicesStoppedInVc[1], "\n "), " ")
+			// checking if current stopped services are same as defined above
+			if reflect.DeepEqual(servicesStoppedInVc, defaultStoppedServicesList) {
+				framework.Logf("All vc required services are in up and running state")
+				noAdditionalServiceStopped = true
+				// wait for 1 min,in case dependent service are still in pending state
+				time.Sleep(1 * time.Minute)
+			} else {
+				for _, service := range servicesStoppedInVc {
+					if !(slices.Contains(defaultStoppedServicesList, service)) {
+						framework.Logf("Starting additional service %s in stopped state", service)
+						sshCmd := fmt.Sprintf("service-control --start %s", service)
```

Comment:
where ever you are doing start service , you can use this method  invokeVCenterServiceControl

Reply:
replaced with method